_inc/client/components/product-expiration/index.jsx

```diff
@@ -31,38 +31,39 @@ class ProductExpiration extends React.PureComponent {
 
 		// Return the subscription date if we don't have the expiry date or the plan is refundable.
 		if ( ! expiryDate || isRefundable ) {
-			const purchaseMoment = moment( purchaseDate );
-			if ( purchaseMoment.isValid() ) {
+			const purchaseDateObj = new Date( purchaseDate );
+			if ( purchaseDateObj.toString() !== 'Invalid Date' ) {
```

Comment:
An alternative here would be `! isNaN( purchaseDateObj.getTime() )`. I don't know which option might be better.

Same below.

Reply:
True, I didn't think of that. I think I'll keep my implementation if that's okay, I find it easier to quickly parse what's going on that way.